ur_robot_driver/include/ur_robot_driver/dashboard_client_ros.h

```diff
@@ -51,5 +56,80 @@ namespace ur_robot_driver
+                          ur_dashboard_msgs::srv::IsProgramRunning::Response::SharedPtr resp);
+
+  bool handleSavedQuery(ur_dashboard_msgs::srv::IsProgramSaved::Request::SharedPtr req,
+                        ur_dashboard_msgs::srv::IsProgramSaved::Response::SharedPtr resp);
+
+  bool handleSafetyModeQuery(ur_dashboard_msgs::srv::GetSafetyMode::Request::SharedPtr req,
+                             ur_dashboard_msgs::srv::GetSafetyMode::Response::SharedPtr resp);
+
+  bool handleRobotModeQuery(ur_dashboard_msgs::srv::GetRobotMode::Request::SharedPtr req,
+                            ur_dashboard_msgs::srv::GetRobotMode::Response::SharedPtr resp);
+
+  bool connect();
+
+  std::shared_ptr<rclcpp::Node> node_;
+  urcl::DashboardClient client_;
+
+  // Commanding services
+  rclcpp::Service<std_srvs::srv::Trigger>::SharedPtr brake_release_service_;
+  rclcpp::Service<std_srvs::srv::Trigger>::SharedPtr clear_operational_mode_service_;
+  rclcpp::Service<std_srvs::srv::Trigger>::SharedPtr close_popup_service_;
+  rclcpp::Service<std_srvs::srv::Trigger>::SharedPtr close_safety_popup_service_;
+  rclcpp::Service<std_srvs::srv::Trigger>::SharedPtr pause_service_;
+  rclcpp::Service<std_srvs::srv::Trigger>::SharedPtr play_service_;
+  rclcpp::Service<std_srvs::srv::Trigger>::SharedPtr power_off_service_;
+  rclcpp::Service<std_srvs::srv::Trigger>::SharedPtr power_on_service_;
+  rclcpp::Service<std_srvs::srv::Trigger>::SharedPtr restart_safety_service_;
+  rclcpp::Service<std_srvs::srv::Trigger>::SharedPtr shutdown_service_;
+  rclcpp::Service<std_srvs::srv::Trigger>::SharedPtr stop_service_;
+  rclcpp::Service<std_srvs::srv::Trigger>::SharedPtr unlock_protective_stop_service_;
+  rclcpp::Service<ur_dashboard_msgs::srv::Load>::SharedPtr load_installation_service_;
+  rclcpp::Service<ur_dashboard_msgs::srv::Load>::SharedPtr load_program_service_;
+  rclcpp::Service<std_srvs::srv::Trigger>::SharedPtr quit_service_;
+  rclcpp::Service<ur_dashboard_msgs::srv::AddToLog>::SharedPtr add_to_log_service_;
+  rclcpp::Service<std_srvs::srv::Trigger>::SharedPtr reconnect_service_;
+  rclcpp::Service<ur_dashboard_msgs::srv::RawRequest>::SharedPtr raw_request_service_;
+
+  // Query services
+  rclcpp::Service<ur_dashboard_msgs::srv::IsProgramRunning>::SharedPtr running_service_;
+  rclcpp::Service<ur_dashboard_msgs::srv::GetLoadedProgram>::SharedPtr get_loaded_program_service_;
+  rclcpp::Service<ur_dashboard_msgs::srv::IsProgramSaved>::SharedPtr is_program_saved_service_;
+  rclcpp::Service<ur_dashboard_msgs::srv::Popup>::SharedPtr popup_service_;
+  rclcpp::Service<ur_dashboard_msgs::srv::GetProgramState>::SharedPtr program_state_service_;
+  rclcpp::Service<ur_dashboard_msgs::srv::GetSafetyMode>::SharedPtr safety_mode_service_;
+  rclcpp::Service<ur_dashboard_msgs::srv::GetRobotMode>::SharedPtr robot_mode_service_;
+
+  // EXISTS AS SPARE IN ROS1 DRIVER
+  //    rclcpp::Service<ur_dashboard_msgs::srv::> polyscope_version_service_;
```

Comment:
I don't understand what this is. Why not just delete it?

Reply:
It existed in the old driver. It is left for possible UR's idea implementation.